server/src/main/java/io/spine/server/event/EventStore.java

```diff
@@ -84,9 +77,34 @@ public static ServiceBuilder newServiceBuilder() {
         return new ServiceBuilder();
     }
 
-    /** Returns default logger for the class. */
-    public static Logger log() {
-        return LogSingleton.INSTANCE.value;
+    private static void checkOfTenant(Event first, Iterator<Event> others) {
+        checkNotNull(first);
+        checkNotNull(others);
+
+        final TenantId firstTenant = getTenantId(first);
+        while (others.hasNext()) {
+            final Event event = others.next();
+            final TenantId currentTenant = getTenantId(event);
+            checkArgument(currentTenant.equals(firstTenant),
+                          DIFFERENT_TENANT_ERROR_TEMPLATE,
+                          first, currentTenant);
+        }
+    }
+
+    /**
+     * Constructs an instance with the passed executor for returning streams.
+     *
+     * @param streamExecutor the executor for updating new subscribers
+     * @param storageFactory the storage factory for creating underlying storage
+     * @param logger         debug logger instance
+     */
+    EventStore(Executor streamExecutor, StorageFactory storageFactory, @Nullable Logger logger) {
```

Comment:
I don't understand why you need this class. There is a builder, which provides the same features.